exporter/datadogexporter/translate_traces.go

```diff
@@ -0,0 +1,492 @@
+
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"go.opencensus.io/trace"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+	tracetranslator "go.opentelemetry.io/collector/translator/trace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+)
+
+// codeDetails specifies information about a trace status code.
+type codeDetails struct {
+	message string // status message
+	status  int    // corresponding HTTP status code
+}
+
+const (
+	deploymentEnv       string = "deployment.environment"
+	keySamplingPriority string = "_sampling_priority_v1"
+	keySpanName         string = "span.name"
+	httpStatusCode      string = "http.status_code"
+	httpMethod          string = "http.method"
+	httpRoute           string = "http.route"
+	versionTag          string = "version"
+	oldILNameTag        string = "otel.instrumentation_library.name"
+	currentILNameTag    string = "otel.library.name"
+)
+
+// statusCodes maps (*trace.SpanData).Status.Code to their message and http status code. See:
+// https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto.
+var statusCodes = map[int32]codeDetails{
+	trace.StatusCodeOK:                 {message: "OK", status: http.StatusOK},
+	trace.StatusCodeCancelled:          {message: "CANCELLED", status: 499},
+	trace.StatusCodeUnknown:            {message: "UNKNOWN", status: http.StatusInternalServerError},
+	trace.StatusCodeInvalidArgument:    {message: "INVALID_ARGUMENT", status: http.StatusBadRequest},
+	trace.StatusCodeDeadlineExceeded:   {message: "DEADLINE_EXCEEDED", status: http.StatusGatewayTimeout},
+	trace.StatusCodeNotFound:           {message: "NOT_FOUND", status: http.StatusNotFound},
+	trace.StatusCodeAlreadyExists:      {message: "ALREADY_EXISTS", status: http.StatusConflict},
+	trace.StatusCodePermissionDenied:   {message: "PERMISSION_DENIED", status: http.StatusForbidden},
+	trace.StatusCodeResourceExhausted:  {message: "RESOURCE_EXHAUSTED", status: http.StatusTooManyRequests},
+	trace.StatusCodeFailedPrecondition: {message: "FAILED_PRECONDITION", status: http.StatusBadRequest},
+	trace.StatusCodeAborted:            {message: "ABORTED", status: http.StatusConflict},
+	trace.StatusCodeOutOfRange:         {message: "OUT_OF_RANGE", status: http.StatusBadRequest},
+	trace.StatusCodeUnimplemented:      {message: "UNIMPLEMENTED", status: http.StatusNotImplemented},
+	trace.StatusCodeInternal:           {message: "INTERNAL", status: http.StatusInternalServerError},
+	trace.StatusCodeUnavailable:        {message: "UNAVAILABLE", status: http.StatusServiceUnavailable},
+	trace.StatusCodeDataLoss:           {message: "DATA_LOSS", status: http.StatusNotImplemented},
+	trace.StatusCodeUnauthenticated:    {message: "UNAUTHENTICATED", status: http.StatusUnauthorized},
+}
+
+// converts Traces into an array of datadog trace payloads grouped by env
+func ConvertToDatadogTd(td pdata.Traces, cfg *Config, globalTags []string) ([]*pb.TracePayload, error) {
```

Comment:
Where do these :-separated global tags come from?

Reply:
These are passed in as a config option in setup https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/187482f5026d08e2dabd92eefd5c45e89bbd3a5c/exporter/datadogexporter/config.go#L97, basically an array of arbitrary list of global tags that user's want to set on every span. So, `"feature_flag:xyz, cloud_provider:alibaba"` or anything along those lines.